tests/test_job_management.py

```diff
@@ -1425,6 +1453,81 @@ def test_email_link_job_resolution(dispatcher_long_living_fixture,
         assert all(key in dict_params_redirect_url for key in dict_param)
 
 
+@pytest.mark.not_safe_parallel
+@pytest.mark.test_catalog
+@pytest.mark.parametrize("catalog_passage", ['file', 'params'])
+@pytest.mark.parametrize("catalog_size", ['short', 'long'])
+def test_email_catalog(dispatcher_long_living_fixture,
+                       dispatcher_local_mail_server,
+                       catalog_passage,
+                       catalog_size):
+    DispatcherJobState.remove_scratch_folders()
+
+    server = dispatcher_long_living_fixture
+    logger.info("constructed server: %s", server)
+
+    # let's generate a valid token
+    token_payload = {
+        **default_token_payload,
+        "roles": "unige-hpc-full, general",
+    }
+    encoded_token = jwt.encode(token_payload, secret_key, algorithm='HS256')
+    selected_catalog_dict = None
+    list_file = None
+    list_file_content = None
+    catalog_object_dict = dict()
+
+    # setting params
+    params = {
+        'query_status': "new",
+        'product_type': 'dummy',
+        'query_type': "Real",
+        'instrument': 'empty-async',
+        'token': encoded_token
+    }
+
+    if catalog_passage == 'file':
+        file_path = DispatcherJobState.create_catalog_file(catalog_value=5, long=(catalog_size == 'long'))
+        list_file = open(file_path)
+        list_file_content = open(file_path).read()
+        catalog_object_dict = BasicCatalog.from_file(file_path).get_dictionary()
+    elif catalog_passage == 'params':
+        catalog_object_dict = DispatcherJobState.create_catalog_object(long=(catalog_size == 'long'))
+        params['selected_catalog'] = json.dumps(catalog_object_dict),
```

Comment:
maybe here try to eliminate all spaces in the string passed. This will break source names, but here it does not matter. I guess it will remain valid json in this case?..

Reply:
Exactly! Also, when setting the catalog within the dispatcher, it is loaded as json, so it will not matter too much the presence/absence of spaces...